app/login/page.tsx

```diff
@@ -1,21 +1,47 @@
+'use client';
+
 import Link from 'next/link';
-import { signIn } from '@/lib/auth';
+// must go through actions.ts file, instead of auth.ts
+import { signInFromLogin } from '@/lib/actions';
 
 import { SimpleLayout } from '@/components/page-layouts';
 import { Label } from '@/components/ui/label';
 import { Button } from '@/components/ui/button';
 import { Input } from '@/components/ui/input';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
 
 export default function Login() {
 
-  const signInAction = async (formData: FormData) => {
-    'use server';
-    await signIn('credentials', {
-      redirectTo: '/',
-      email: formData.get('email') as string,
-      password: formData.get('password') as string,
-    });
-  };
+  const schema = z.object({
+    email: z.string().trim().min(1, {message: "Email is required"}).email(),
```

Comment:
Looking at this again, I'm not sure if the trim actually changes the data being sent to the database or if it just does the trim for the validation. Could you double check this? It would be odd if there was an email in the database that started with a space or something

Reply:
Checked and trim doesn't change the data being sent to the database - removed from both register & login pages